NEWS.md

```diff
@@ -63,6 +63,8 @@ rowwiseDT(
     # 15: All values  Total  Total   999   999   NaN    10
     ```
 
+4. `patterns()` returns character instead of integer vector(s), to support user-defined `cols`, which can be useful to specify a subset of columns to reshape without having to use a regex, [#6498](https://github.com/Rdatatable/data.table/issues/6498). Thanks to @hongyuanjia for the report, and to @tdhock for the PR.
```

Comment:
Since `patterns()` is kind-of "internal", I don't think we need to explain the difference of returning character vs. integer (it's sort of an implementation detail). I think it suffices to just explain the user-facing behavior change.

Implementation change could be described in PR description as a hint to code reviewer.
```suggestion
4. `patterns()` in `melt()` combines correctly with user-defined `cols=`, which can be useful to specify a subset of columns to reshape without having to use a regex, for example `patterns("2", cols=c("y1", "y2"))` will only give `y2` even if there are other columns in the input matching `2`, [#6498](https://github.com/Rdatatable/data.table/issues/6498). Thanks to @hongyuanjia for the report, and to @tdhock for the PR.
```